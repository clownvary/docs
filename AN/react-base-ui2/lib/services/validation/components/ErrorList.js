'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var


ErrorList = function (_PureComponent) {(0, _inherits3.default)(ErrorList, _PureComponent);function ErrorList() {(0, _classCallCheck3.default)(this, ErrorList);return (0, _possibleConstructorReturn3.default)(this, (ErrorList.__proto__ || (0, _getPrototypeOf2.default)(ErrorList)).apply(this, arguments));}(0, _createClass3.default)(ErrorList, [{ key: 'render', value: function render()
    {var _props =
      this.props,messages = _props.messages,className = _props.className,props = (0, _objectWithoutProperties3.default)(_props, ['messages', 'className']);
      return (
        _react2.default.createElement('div', (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('an-error-list', className) }),
          messages.map(function (msg, i) {return (
              _react2.default.createElement('div', { key: i, className: 'an-error' },
                _react2.default.createElement('span', { className: 'icon aaui-alert-error-icon icon-times-circle' }),
                _react2.default.createElement('span', null, msg)));})));






    } }]);return ErrorList;}(_react.PureComponent);exports.default = ErrorList;module.exports = exports['default'];