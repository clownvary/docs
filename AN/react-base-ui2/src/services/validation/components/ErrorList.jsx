import React, { PureComponent } from 'react';
import classNames from 'classnames';


export default class ErrorList extends PureComponent {
  render() {
    const { messages, className, ...props } = this.props;
    return (
      <div {...props} className={classNames('an-error-list', className)}>
        { messages.map((msg, i) => (
          <div key={i} className="an-error">
            <span className="icon aaui-alert-error-icon icon-times-circle" />
            <span>{msg}</span>
          </div>
          )
        )
      }
      </div>
    );
  }
}
