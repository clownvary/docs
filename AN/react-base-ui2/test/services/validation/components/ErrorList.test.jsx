import React from 'react';
import toJSON from 'enzyme-to-json';
import { mount } from 'enzyme';
import ErrorList from 'src/services/validation/components/ErrorList';

describe('services/validation/components/ErrorList.jsx', () => {
  test('rendering', () => {
    const wrapper = mount(<ErrorList messages={['test messages']} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
