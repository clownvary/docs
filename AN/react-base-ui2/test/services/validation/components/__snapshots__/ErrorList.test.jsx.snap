// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services/validation/components/ErrorList.jsx rendering 1`] = `
<ErrorList
  messages={
    Array [
      "test messages",
    ]
  }
>
  <div
    className="an-error-list"
  >
    <div
      className="an-error"
      key="0"
    >
      <span
        className="icon aaui-alert-error-icon icon-times-circle"
      />
      <span>
        test messages
      </span>
    </div>
  </div>
</ErrorList>
`;
